cmd/sbotcli/main.go

```diff
@@ -307,6 +308,65 @@ var sourceCmd = &cli.Command{
 	},
 }
 
+var getSubsetCmd = &cli.Command{
+	Name:  "subset",
+	Usage: "invoke the partialReplication.getSubset muxrpc",
+
+	// define cli flags
+	Flags: []cli.Flag{
+		&cli.IntFlag{Name: "limit", Value: -1},
+		&cli.BoolFlag{Name: "dsc", Value: false},
+		&cli.BoolFlag{Name: "keys", Value: false},
+	},
+
+	Action: func(ctx *cli.Context) error {
+		if ctx.NArg() < 1 {
+			return errors.New(`subset usage: sbotcli subset [optional --flags] <valid json>`)
+		}
+		input := ctx.Args().Get(0)
+		cmd := "subset"
+
+		client, err := newClient(ctx)
+		if err != nil {
+			return err
+		}
+
+		// a helper struct borrowed from the query package for converting the string input into a something we can send over
+		// the wire & which will be unpacked correctly
+		var payload struct {
+			Operation string `json:"op"`
+
+			Args   []query.SubsetOperation `json:"args,omitempty"`
+			String string            `json:"string,omitempty"`
+			Feed   *refs.FeedRef     `json:"feed,omitempty"`
+			// embed query.SubsetOptions (adds limit, dsc and keys)
+			query.SubsetOptions
```

Comment:
I think this needs to be a 2nd argument when calling the rpc.

i.e. `[{query....}, {options...}]`

Reply:
ah! i was looking everywhere for the function declaration of the source call and had the hardest time finding it; thanks for the tip!

is this it? or is there an interface somewhere that i'm missing out on https://github.com/cryptoscope/go-muxrpc/blob/master/stream2_source.go

Reply:
interface for..?

The arguments for a call are handled equally for all types of a call (async, duplex, source, sink). In go-land they end up on the [RawArgs](https://github.com/cryptoscope/ssb/search?q=RawArgs) field of the Request struct. Which RPC expects which count and format of the arguments is handled individually by each handler method.